### PR TITLE
Tests for aisim's modules

### LINKED ISSUE #39: More tests
We should make tests for testing the internal structure of the modules.
My first two cents for tests of the AtomicEnsemble class:
```
N_atom = 1000
N_ints = [2, 3, 5, 10, 20, 100]
for N_int in N_ints:
    print(N_int)
    # Generate random states
    random_phase_space_vectors = np.random.rand(N_atom, 6)
    random_kets = np.random.rand(N_atom, N_int)
    # normalize random ket
    norm = np.einsum('ij,ij->i', random_kets, random_kets)
    norm_random_kets = np.einsum('ij,i->ij', random_kets, 1/norm**(1/2))
    # apply random phase shifts
    random_phases = 2*np.pi*np.random.rand(N_atom, N_int)
    norm_random_kets = np.exp(1j*random_phases) * norm_random_kets
    norm_random_kets = np.reshape(norm_random_kets, (N_atom, N_int, 1))
    random_atoms = ais.AtomicEnsemble(
        random_phase_space_vectors, norm_random_kets)
    # Testing AtomicEnsemble properties
    # Test duality of Kets and Bras
    np.testing.assert_array_almost_equal(random_atoms.state_kets, np.conjugate(
        np.transpose(random_atoms.state_bras, (0, 2, 1))))
    # Test whether the trace of every density matrix is equal to one
    np.testing.assert_almost_equal(
        np.trace(random_atoms.density_matrices, axis1=1, axis2=2), 1)
    # Test whether every single atoms density matrix satisfies rho^2 = rho (condition for pure states)
    np.testing.assert_array_almost_equal(np.matmul(
        random_atoms.density_matrices, random_atoms.density_matrices), random_atoms.density_matrices)
```

### PR DESCRIPTION
**Pull request by savowe**
closes #39

### CODE REVIEW

File: aisim/atoms.py [hunk 1]
```diff
@@ -208,6 +212,9 @@ def create_random_ensemble_from_gaussian_distribution(pos_params, vel_params,
     ensemble : AtomicEnsemble
         Atomic ensemble containing the generated phase space vectors.
     """
+    if seed is not None:
```
Comment (savowe) [inline aisim/atoms.py:215]: I think the seed function needs to be called after every single instance of generating random numbers.
Unfortunately, the seed just resets itself after generating a random

Comment (bleykauf) [reply aisim/atoms.py:215]: After you set numpy.random.seed all random numbers generated after that are predictable:

```python
import numpy as np

np.random.seed(0)
rand1 = np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand2 = np.array([np.random.normal() for i in range(10)])

rand1 == rand2
```

gives:

```python
array([ True,  True,  True,  True,  True,  True,  True,  True,  True, True])
```

Comment (bleykauf) [reply aisim/atoms.py:215]: The setting is also global I think, so instead of passing `seed` you could also simply call `np.random.seed()`.

**Comment by bleykauf** [comment]
Just checked what is still missing with: ` python -m pytest --cov=aisim --cov-report term-missing`:

```
----------- coverage: platform linux, python 3.7.6-final-0 -----------
Name                Stmts   Miss  Cover   Missing
-------------------------------------------------
aisim/__init__.py       8      0   100%
aisim/atoms.py         84      5    94%   134-137, 184, 317
aisim/beam.py          77      0   100%
aisim/convert.py       27      0   100%
aisim/det.py           11      0   100%
aisim/prop.py          47     27    43%   30, 33, 99, 105-149
-------------------------------------------------
TOTAL                 531    184    65%
```

I am not sure if line 317 in atoms.py is still necessary.

**Comment by bleykauf** [comment]
Ok, so only lines 134-137, 184, 317 in `atoms.py` are missing...

File: aisim/atoms.py (hunk 1 above)
Comment (bleykauf) [reply aisim/atoms.py:215]: @savowe : can you confirm?

Comment (savowe) [reply aisim/atoms.py:215]: @bleykauf  yes. i did not think that also a second funciton call would give the same results but i tried
```
np.random.seed(0)
rand11 = np.array([np.random.normal() for i in range(10)])
rand12 = np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand21 = np.array([np.random.normal() for i in range(10)])
rand22 = np.array([np.random.normal() for i in range(10)])

rand11 == rand21
rand12 == rand22
```
gives
```
>>> rand11 == rand21
array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True])
>>> rand12 == rand22
array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True])
```
So I now think calling the functions once is good!.
If you put the random number generator inside a new function
```
def call_random(): return np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand11 = call_random()
rand12 = call_random()

np.random.seed(0)
rand21 = call_random()
rand22 = call_random()

rand11 == rand21
rand12 == rand22
```
 it also works. So the scope goes at least into the function. I dont know whether it is now considered global

**Comment by bleykauf** [comment]
@savowe : please don't forget du squash before merging

**Comment by savowe** [comment]
you go

**Pull request merged**